src/jitlog/reader.lua

```diff
@@ -167,7 +167,24 @@ function logreader:processheader(header)
   end
   
   self.msgsizes = header.msgsizes
-  for _, size in ipairs(header.msgsizes) do
+  for i, size in ipairs(header.msgsizes) do
+    local name = header.msgnames[i]
+    local id = MsgType[name]
+    if id and msgsizes[id + 1] ~= size then
+      local oursize = msgsizes[id + 1]
```

Comment:
I think this would be clearer as:

```lua
local oursize = math.abs(msgsizes[id + 1])
```

(ditto the line below) rather than the tedious manual `if < 0` dance.